src/core/autosave/autosave.business.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  DatabaseSchemaVm,
+  createDefaultDatabaseSchemaVm,
+} from '@/core/providers';
+
+export interface CanvasSchema {
+  filename: string | undefined;
+  canvasSchema: DatabaseSchemaVm;
+}
+
+export const saveToLocal = (key: string, value: CanvasSchema) => {
+  try {
+    const deepCopyValue = JSON.parse(JSON.stringify(value));
+    const stringifiedDeepCopyValue = JSON.stringify(deepCopyValue);
+    localStorage.setItem(key, stringifiedDeepCopyValue);
+  } catch (error) {
+    console.error('Autosave error', error);
+    setTimeout(() => {
+      saveToLocal(key, value);
+    }, 30000);
+  }
+};
+
+export const retrieveFromLocal = (
+  setLoadSample: (T: boolean) => void
+): DatabaseSchemaVm => {
+  try {
+    const value = localStorage.getItem('autoSaveFile');
```

Comment:
Check if we can use the loca storage commo wrapper, if not expand functionallity on our common stuff